support/k8s/k8s-workload-registrar/controller_test.go

```diff
@@ -437,96 +440,146 @@ func requireReviewAdmissionFailure(t *testing.T, controller *Controller, req *ad
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	entry, ok := c.entries[id.Id]
-	if !ok {
-		return nil, fmt.Errorf("entry %q not found", id.Id)
+	resp := new(entryv1.BatchDeleteEntryResponse)
+	for _, id := range req.Ids {
+		_, ok := c.entries[id]
+		code := codes.OK
+		var msg string
+		if !ok {
+			code = codes.NotFound
+			msg = "not found"
+		}
+
+		resp.Results = append(resp.Results, &entryv1.BatchDeleteEntryResponse_Result{
+			Status: &types.Status{Code: int32(code), Message: msg},
+			Id:     id,
+		})
+		delete(c.entries, id)
 	}
-	delete(c.entries, id.Id)
 
-	return entry, nil
+	return resp, nil
 }
 
-func (c *fakeRegistrationClient) ListBySelectors(ctx context.Context, selectors *common.Selectors, opts ...grpc.CallOption) (*common.RegistrationEntries, error) {
+func (c *fakeEntryClient) ListEntries(ctx context.Context, req *entryv1.ListEntriesRequest, opts ...grpc.CallOption) (*entryv1.ListEntriesResponse, error) {
+	switch {
+	case req.Filter == nil:
+		return nil, status.Error(codes.InvalidArgument, "expecting filter")
+	case req.Filter.BySelectors == nil:
+		return nil, status.Error(codes.InvalidArgument, "expecting filter by selector")
+	case req.Filter.BySelectors.Match != types.SelectorMatch_MATCH_EXACT:
+		return nil, status.Error(codes.InvalidArgument, "expecting exact selector match")
+	}
+
 	// peform an exact match check against selectors
-	var entries []*common.RegistrationEntry
+	var entries []*types.Entry
 	for _, entry := range c.entries {
-		if areSelectorsEqual(selectors.Entries, entry.Selectors) {
-			entries = append(entries, cloneRegistrationEntry(entry))
+		if selectorSetsEqual(req.Filter.BySelectors.Selectors, entry.Selectors) {
+			entries = append(entries, cloneEntry(entry))
 		}
 	}
 
-	return &common.RegistrationEntries{
+	return &entryv1.ListEntriesResponse{
 		Entries: entries,
 	}, nil
 }
 
-func requireEntriesEqual(t *testing.T, expected, actual []*common.RegistrationEntry) {
-	actual = cloneRegistrationEntries(actual)
-	util.SortRegistrationEntries(actual)
-	expected = cloneRegistrationEntries(expected)
-	util.SortRegistrationEntries(expected)
-	require.Equal(t, expected, actual)
+func requireEntriesEqual(t *testing.T, expected, actual []*types.Entry) {
+	spiretest.RequireProtoListEqual(t, expected, actual)
 }
 
-func areSelectorsEqual(expected, actual []*common.Selector) bool {
-	actual = cloneSelectors(actual)
-	util.SortSelectors(actual)
-	expected = cloneSelectors(expected)
-	util.SortSelectors(expected)
-	return proto.Equal(&common.Selectors{Entries: actual}, &common.Selectors{Entries: expected})
+func selectorSetsEqual(as, bs []*types.Selector) bool {
+	if len(as) != len(bs) {
+		return false
+	}
+	type sel struct {
+		t string
+		v string
+	}
+	set := map[sel]struct{}{}
+	for _, a := range as {
+		set[sel{t: a.Type, v: a.Value}] = struct{}{}
+	}
+	for _, b := range bs {
+		if _, ok := set[sel{t: b.Type, v: b.Value}]; !ok {
+			return false
+		}
+	}
+	return true
```

Comment:
This should probably be a common library function.

Reply:
Yes, totally agreed. I punted on doing that now. As part of a refactor of the agent (related to #1847), I'm introducing some high-level types for things like entries and selectors. The new types include a `Selectors` type that will include set operations over a selector set. These types will be available as part of a higher level library for the Entry (and other) APIs.